cryptoki/src/types/mechanism/ecdh.rs

```diff
@@ -0,0 +1,31 @@
+//! ECDH mechanism types
+
+use crate::types::Ulong;
+use cryptoki_sys::CK_EC_KDF_TYPE;
+use std::ffi::c_void;
+
+/// ECDH derivation parameters.
+///
+/// The elliptic curve Diffie-Hellman (ECDH) key derivation mechanism
+/// is a mechanism for key derivation based on the Diffie-Hellman
+/// version of the elliptic curve key agreement scheme, as defined in
+/// ANSI X9.63, where each party contributes one key pair all using
+/// the same EC domain parameters.
+///
+/// This structure wraps CKM_ECDH1_DERIVE mechanism.
+#[derive(Copy, Debug, Clone)]
+#[repr(C)]
+pub struct EcdhDeriveParams {
```

Comment:
What is the 1 for? Personally I'd vote for improving naming in a Rusty lib if possible :)

Reply:
The 1 is coming from the PKCS11 name `CK_ECDH1_DERIVE_PARAMS`. I guess we can discuss this, my idea of a Rust abstraction crate, at this level, is to be a systematic translation to an idiomatic and safe Rust library where there is a 1-1 equivalent to everything in the C library. I thought we would keep the 1 here just to be consitent with how we have been translating the PKCS 11 names to the Rust ones, without trying to be more "clever" than that.
I guess they put a 1 in case they write a new version of this struct in a later version?

Reply:
> I guess they put a 1 in case they write a new version of this struct in a later version?

At the same time, I doubt there will be another 2.xx version coming out, so if there isn't a new version of the struct now, it shall forever be like this.

Reply:
Yeah, to be honest initially when I didn't know PKCS#11 using this crate was kind of hard. But the more I dug PKCS#11 I see the value in staying consistent with the spec. For example when I added more constants it was easy to follow which parts of the spec needs to be consulted.

So: I think it's worth to keep consistent with PKCS#11 here but have another crate that provides more high level interface (like parsec?).